src/Utils.php

```diff
@@ -759,6 +773,66 @@ public static function check_products_for_letterbox( $products ) {
 		return $has_letterbox_product && $total_ratio_letterbox_item <= 1;
 	}
 
+	/**
+	 * Check if current order is for adult.
+	 *
+	 * @param \WC_Order|int $order \WC_order or Order ID.
+	 *
+	 * @return boolean
+	 */
+	public static function is_order_for_adults( $order ): bool {
+		if ( 'BE' === wc_get_base_location()['country'] ) {
+			return false;
+		}
+
+		// Check if order id provided.
+		if ( is_int( $order ) ) {
+			$order = wc_get_order( $order );
+		}
+
+		if ( ! is_a( $order, 'WC_Order' ) ) {
+			return false;
+		}
+
+		if ( ! in_array( $order->get_shipping_country(), self::get_available_country_for_adult(), true ) ) {
+			return false;
+		}
+
+		$products = $order->get_items();
+		$is_adult = self::check_products_for_adults( $products );
+
+		return $is_adult;
+	}
+
+	/**
+	 * Check if any product is for adult.
+	 *
+	 * @param array $products WC_Products[] or order_item[].
+	 *
+	 * @return bool
+	 */
+	public static function check_products_for_adults( $products ): bool {
+
+		foreach ( $products as $item_id => $item ) {
+			$product = wc_get_product( $item['product_id'] ?? $item->get_product_id() );
+			if ( ! is_a( $product, 'WC_Product' ) ) {
+				continue;
+			}
+
+			if ( ! $product->needs_shipping() ) {
+				continue;
+			}
+
+			$is_adult_product = $product->get_meta( Single::ADULTS_ONLY_FIELD );
+
+			if ( 'yes' === $is_adult_product ) {
+				return true;
+			}
```

Comment:
I suggest to extract this following logic into a small utilityfunction for better readability and reuse.
```php
/**
 * Check if the given product is marked as adults-only.
 *
 * @param WC_Product $product Product object.
 * @return bool
 */
public static function is_product_adults_only( WC_Product $product ): bool {
	return 'yes' === $product->get_meta( Single::ADULTS_ONLY_FIELD );
}
```